1-js/09-classes/01-class/article.md

```diff
@@ -50,87 +50,89 @@ user.sayHi();
+نماد در اینجا نباید با شیء لفظی اشتباه شود. در کلاس، نیازی به کاما نیست.
 ```
 
-## What is a class?
+## یک کلاس چیست؟
 
-So, what exactly is a `class`? That's not an entirely new language-level entity, as one might think.
+بنابراین، `class` دقیقاً چیست؟ این یک موجودیت کاملاً جدید در سطح زبان نیست، همانطور که ممکن است تصور شود.
 
-Let's unveil any magic and see what a class really is. That'll help in understanding many complex aspects.
+بیایید از هر جادویی رونمایی کنیم و ببینیم کلاس واقعاً چیست. این به درک بسیاری از جنبه‌های پیچیده کمک می‌کند.
 
-In JavaScript, a class is a kind of function.
+در جاوااسکریپت کلاس نوعی تابع است.
 
-Here, take a look:
+در اینجا، نگاهی بیندازیم:
 
 ```js run
 class User {
   constructor(name) { this.name = name; }
   sayHi() { alert(this.name); }
 }
 
-// proof: User is a function
+// اثبات: کاربر یک تابع است
 *!*
-alert(typeof User); // function
+alert(typeof User); // تابع
 */!*
 ```
 
-What `class User {...}` construct really does is:
+کاری که ساختار `class User {...}` واقعا انجام می‌دهد این است:
 
-1. Creates a function named `User`, that becomes the result of the class declaration. The function code is taken from the `constructor` method (assumed empty if we don't write such method).
-2. Stores class methods, such as `sayHi`, in `User.prototype`.
+1. تابعی به نام `User` ایجاد می‌کند که نتیجه تعریف کلاس می‌شود. کد تابع از متد سازنده گرفته شده است (اگر چنین روشی را ننویسیم خالی فرض می‌شود).
+2. متدهای کلاس، مانند `sayHi` را در `User.prototype` ذخیره می‌کند.
 
-After `new User` object is created, when we call its method, it's taken from the prototype, just as described in the chapter <info:function-prototype>. So the object has access to class methods.
+پس از ایجاد شیء `new User`، وقتی متد آن را فراخوانی می‌کنیم، از نمونه اولیه گرفته می‌شود، درست همانطور که در فصل <info:function-prototype> توضیح داده شد. بنابراین شیء به متدهای کلاس دسترسی دارد.
 
-We can illustrate the result of `class User` declaration as:
+ما می‌توانیم نتیجه تعریف `class User` را به صورت زیر نشان دهیم:
 
 ![](class-user.svg)
 
-Here's the code to introspect it:
+این کد برای درون‌یابی آن است:
 
 ```js run
 class User {
   constructor(name) { this.name = name; }
   sayHi() { alert(this.name); }
 }
 
-// class is a function
-alert(typeof User); // function
+// کلاس یک تابع است
+alert(typeof User); // تابع
 
-// ...or, more precisely, the constructor method
+// ... یا به طور دقیق تر، متد سازنده
 alert(User === User.prototype.constructor); // true
 
-// The methods are in User.prototype, e.g:
-alert(User.prototype.sayHi); // the code of the sayHi method
+// :هستند، به عنوان مثال User.prototype متدها در
+alert(User.prototype.sayHi); // sayHi کد متد
 
-// there are exactly two methods in the prototype
-alert(Object.getOwnPropertyNames(User.prototype)); // constructor, sayHi
+// دقیقا دو متد در پروتوتایپ وجود دارد
+alert(Object.getOwnPropertyNames(User.prototype)); // sayHi ،سازنده
 ```
 
-## Not just a syntactic sugar
+## فقط یک سینتکس شکری نیست
```

Comment:
I'm not sure about this translation, Let me know if it's wrong.

Reply:
```suggestion
## فقط یک سینتکس برای زیباکردن(syntactic sugar) نیست
```